dec/decode.go

```diff
@@ -122,7 +127,7 @@ func (r *BrotliReader) Read(p []byte) (cnt int, err error) {
 				&availableOut,
 				&nextOut,
 				&r.totalOut,
-				r.state,
+				(*C.BrotliState)(unsafe.Pointer(&r.state[0])),
```

Comment:
this cast might be overly long-winded, I have no idea what a typical gopher would write